test/parallel/test-repl-context.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const repl = require('repl');
+
+// Create a dummy stream that does nothing
+const stream = new common.ArrayStream();
+
+// Test when useGlobal is false
+testContext(repl.start({
+  input: stream,
+  output: stream,
+  useGlobal: false
+}));
+
+function testContext(repl) {
+  const context = repl.createContext();
+  // ensure that the repl context gets its own "console" instance
+  assert(context.console instanceof require('console').Console);
+
+  // ensure that the repl's global property is the context
+  assert(context.global === context);
+  assert(context.global.global === context);
```

Comment:
Well, to be honest, I'm not sure why the original `repl.js` did `context.global.global = context` [here](https://github.com/nodejs/node/blob/master/lib/repl.js#L591). But I've replicated that behavior and am testing that I've actually done it here.